File: api/src/endpoint/question/create_answer_comment.rs
use api_derive::ApiRequest;
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::question::CommentResource};

pub struct CreateAnswerComment;

impl Endpoint for CreateAnswerComment {
    const PATH: &'static str =
        "/repository/{owner}/{repo}/question/{number}/answer/{answer_id}/comment";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateAnswerCommentRequest;
    type Response = CreateAnswerCommentResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct CreateAnswerCommentRequest {
    pub body: String,
}

pub type CreateAnswerCommentResponse = CommentResource;

File: gitdot-api-ts/src/endpoint/question/create-answer-comment.ts
import { z } from "zod";

import { CommentResource } from "../../resource";

export const CreateAnswerCommentRequest = z.object({
  body: z.string(),
});
export type CreateAnswerCommentRequest = z.infer<
  typeof CreateAnswerCommentRequest
>;

export const CreateAnswerComment = {
  path: "/repository/{owner}/{repo}/question/{number}/answer/{answer_id}/comment",
  method: "POST",
  request: CreateAnswerCommentRequest,
  response: CommentResource,
} as const;
export type CreateAnswerComment = typeof CreateAnswerComment;
